scipio/src/executor.rs

```diff
@@ -322,28 +352,99 @@ impl LocalExecutor {
         };
 
-        if let Some(cpu) = binding {
-            bind_to_cpu(cpu)?;
-        }
+        le.init()?;
+        Ok(le)
+    }
 
-        let queues = le.queues.clone();
-        let index = 0;
+    /// Creates a single-threaded executor, optionally bound to a specific CPU, inside
+    /// a newly craeted thread. The parameter `name` specifies the name of the thread.
+    ///
+    /// This is a more ergonomic way to create a thread and then run an executor inside it
+    /// This function panics if creating the thread or the executor fails. If you need more
+    /// fine-grained error handling consider initializing those entities manually.
+    ///
+    /// You must call wait_on_executors() to wait for all executors spawned through spawn_new
+    /// to return
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::LocalExecutor;
+    ///
+    /// // executor is a single thread, but not bound to any particular CPU.
+    /// LocalExecutor::spawn_new("myname", None, async move {
+    ///     println!("hello");
+    /// });
+    ///
+    /// LocalExecutor::wait_on_executors();
+    /// ```
+    pub fn spawn_new<F, T>(name: &'static str, binding: Option<usize>, fut: F)
+    where
+        F: Send + 'static,
+        F: Future<Output = T>,
+        T: Send + 'static,
+    {
+        let id = EXECUTOR_ID.fetch_add(1, Ordering::Relaxed);
+
+        let mut executors = SPAWNED_EXECUTORS.lock().unwrap();
+        (*executors).push(
+            Builder::new()
+                .name(format!("{}-{}", name, id).to_string())
+                .spawn(move || {
+                    let mut le = LocalExecutor {
+                        queues: ExecutorQueues::new(),
+                        parker: parking::Parker::new(),
+                        binding,
+                        id,
+                    };
+                    le.init().unwrap();
+                    le.run(async move {
+                        let task = Task::local(async move {
+                            fut.await;
+                        });
+                        task.await;
+                    })
+                })
+                .unwrap(),
+        )
+    }
 
-        let io_requirements = IoRequirements::new(Latency::NotImportant, 0);
-        le.queues.borrow_mut().available_executors.insert(
-            0,
-            TaskQueue::new("default", 1000, io_requirements, move || {
-                let mut queues = queues.borrow_mut();
-                queues.maybe_activate(index);
-            }),
-        );
-        Ok(le)
+    /// Wait for all executors called through spawn_new to return
+    ///
+    /// # Examples
+    /// use scipio::LocalExecutor;
+    ///
+    /// // executor is a single thread, but not bound to any particular CPU.
+    /// LocalExecutor::spawn_new("myname", None, async move {
+    ///     println!("hello");
+    /// });
+    ///
+    /// LocalExecutor::wait_on_executors();
+    pub fn wait_on_executors() {
```

Comment:
Seems like a nice utility but I see a couple of problems with it:

- people know how to use threads and join them. This function + the void `spawn_new` make something simple complicated (at least, specific to this library);
- it prevents you from from stopping only part of your executors.

Reply:
The reason I didn't do it this way is that it forces you to have a vector to store the handles. In the search for ergonomics that's one more thing to care about.

It's not the end of the world. I will consider it.

Reply:
changed.